float64 offset1
float64 offset2
float64 offset3
float64 offset4

string[] paramValue
---

int32 res